neo4j-store/src/main/kotlin/org/ostelco/prime/storage/graph/Neo4jStore.kt

```diff
@@ -806,26 +806,43 @@ object Neo4jStoreSingleton : GraphStore {
         }
 
         return IO {
-            Either.monad<StoreError>()
-                    .binding {
-                        simProfiles.bind().map { simProfile ->
-                            val regionId = (regionCode ?: map[simProfile.id])
-                                    ?: ValidationError(type = simProfileEntity.name, id = simProfile.iccId, message = "SimProfile not linked to any region")
-                                            .left()
-                                            .bind()
-
-                            val simEntry = simManager.getSimProfile(
-                                    hlr = hssNameLookup.getHssName(regionId),
-                                    iccId = simProfile.iccId)
-                                    .mapLeft { NotFoundError(type = simProfileEntity.name, id = simProfile.iccId) }
-                                    .bind()
-                            org.ostelco.prime.model.SimProfile(
-                                    iccId = simProfile.iccId,
-                                    alias = simProfile.alias,
-                                    eSimActivationCode = simEntry.eSimActivationCode,
-                                    status = simEntry.status)
-                        }
-                    }.fix()
+            Either.monad<StoreError>().binding {
+                simProfiles.bind().map { simProfile ->
+
+                    val regionId = (regionCode ?: map[simProfile.id])
+
+                    val simEntry = if (regionId != null) {
+                        simManager.getSimProfile(
+                                hlr = hssNameLookup.getHssName(regionId),
+                                iccId = simProfile.iccId).fold(
+                                { error ->
+                                    logger.warn("SimProfile not found in SIM Manager DB. region: {}, iccId: {}, error: {}", regionId, simProfile.iccId, error)
+                                    SimEntry(
+                                            iccId = simProfile.iccId,
```

Comment:
This looks like repeated code (the same identical code exists below). Repeated code like this makes the code more verbose than it should.  Can this repetition  be replaced by a pre-computed value or function generating the value on the fly?